hikari/impl/entity_factory.py

```diff
@@ -2177,6 +2184,56 @@ def deserialize_autocomplete_interaction(
             guild_locale=locales.Locale(payload["guild_locale"]) if "guild_locale" in payload else None,
         )
 
+    def deserialize_modal_interaction(self, payload: data_binding.JSONObject) -> modal_interactions.ModalInteraction:
+        data_payload = payload["data"]
+
+        guild_id: typing.Optional[snowflakes.Snowflake] = None
+        if raw_guild_id := payload.get("guild_id"):
+            guild_id = snowflakes.Snowflake(raw_guild_id)
+
+        member: typing.Optional[base_interactions.InteractionMember]
+        if member_payload := payload.get("member"):
+            assert guild_id is not None
+            member = self._deserialize_interaction_member(member_payload, guild_id=guild_id)
+            # See https://github.com/discord/discord-api-docs/pull/2568
+            user = member.user
+
+        else:
+            member = None
+            user = self.deserialize_user(payload["user"])
+
+        app_perms = payload.get("app_permissions")
+
+        components: typing.List[modal_interactions.PartialModalComponent] = []
+        for component_payload in data_payload["components"]:
+            try:
+                components.append(self._deserialize_modal_component(component_payload))
+            except errors.UnrecognisedEntityError:
+                pass
+
+        message: typing.Optional[message_models.Message] = None
+        if message_payload := payload.get("message"):
+            message = self.deserialize_message(message_payload)
+
+        return modal_interactions.ModalInteraction(
+            app=self._app,
+            application_id=snowflakes.Snowflake(payload["application_id"]),
+            id=snowflakes.Snowflake(payload["id"]),
+            type=base_interactions.InteractionType(payload["type"]),
+            guild_id=guild_id,
+            app_permissions=permission_models.Permissions(app_perms) if app_perms is not None else None,
```

Comment:
```suggestion
            app_permissions=permission_models.Permissions(app_perms) if app_perms else None,
```

Reply:
Can the perms not be `0`?

Reply:
It will be `"0"` tho, which will pass the check. Also, just realized that you are missing a cast to `int` before passing it to the enum